test/integrations/testUtils.ts

```diff
@@ -104,6 +97,42 @@ export const overrideDestination = (destination: Destination, overrideConfigValu
   });
 };
 
+export const produceTestData = (testData: TestCaseData[]) => {
+  const result: any = [];
+  testData.forEach((tcData) => {
+    let events;
+    try {
+      switch (tcData.feature) {
+        case tags.FEATURES.PROCESSOR:
+          events = tcData.input.request.body;
+          break;
+        case tags.FEATURES.BATCH:
+          events = tcData.input.request.body.input;
+          break;
+        case tags.FEATURES.ROUTER:
+          events = tcData.input.request.body.input;
+          break;
+      }
+    } catch (e) {
+      throw new Error(`Error in producing test data: ${e}`);
```

Comment:
capture which destination or source causing this issue and make test ID

Reply:
addressed